qiskit_addon_sqd/subsampling.py

```diff
@@ -136,7 +135,8 @@ def subsample(
     batches = []
     for _ in range(num_batches):
         if randomly_sample:
-            indices = np.random.choice(
+            rng = np.random.default_rng(rand_seed)
```

Comment:
The `rng` should always be initialized only once, at the beginning of the function. If it's initialized more than once in the function (like here), that's a bug.